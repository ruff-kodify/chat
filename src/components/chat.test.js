import React from 'react';
import Chat from './chat';

test('it should render properly', () => {
  const wrapper = shallow(<Chat />);
  expect(wrapper).toMatchSnapshot();
});

test('it should add a thought', () => {
  const wrapper = mount(<Chat />);
  const input = wrapper.find('[data-test="input"]');
  const button = wrapper.find('[data-test="button"]');
  const arg = 'Oops I did it again';

  expect(wrapper.find('.message--thought')).toHaveLength(0);

  input.simulate('change', { target: { value: `/think "${arg}"` } });
  button.simulate('click');

  expect(wrapper.find('.message--thought')).toHaveLength(1);
  expect(wrapper.find('.message--thought').text()).toBe(arg);
});

test('it should add a new user message', () => {
  const wrapper = mount(<Chat />);
  const input = wrapper.find('[data-test="input"]');
  const button = wrapper.find('[data-test="button"]');
  const message = 'Hello';

  expect(wrapper.find('.message')).toHaveLength(0);

  input.simulate('change', { target: { value: message } });
  button.simulate('click');

  expect(wrapper.find('.message')).toHaveLength(1);
  expect(wrapper.find('.message--user')).toHaveLength(1);
});
