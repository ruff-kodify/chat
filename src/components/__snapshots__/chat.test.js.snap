// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render properly 1`] = `
<div
  className="chat"
>
  <PartnerInfo />
  <Messages
    messages={Array []}
  />
  <Input
    onSend={[Function]}
    onType={[Function]}
  />
</div>
`;
